src/backtesting/simulator.py

```
@@ -0,0 +1,242 @@
+        
+        logger.debug(f"SELL: {shares_to_sell:.2f} shares of {ticker} at ${execution_price:.2f}")
+    
+    def _calculate_portfolio_value(self, current_data: pd.Series, ticker: str) -> float:
+        """Calculate current portfolio value."""
+        stock_value = 0
+        if ticker in self.positions and self.positions[ticker] > 0:
+            stock_value = self.positions[ticker] * current_data['Close']
+        
+        return self.cash + stock_value
+    
+    def _calculate_performance_metrics(self, data: pd.DataFrame, ticker: str) -> Dict:
```

Pylint (3.3.7)

[refactor] Too many local variables (28/15)

(R0914)